internal/commands/pull/inputs.go

```diff
@@ -109,3 +118,46 @@ func (i *inputs) resolveRemoteApp(ui terminal.UI, clients cli.Clients) (realm.Ap
 
 	return app, nil
 }
+
+func (i *inputs) resolveTemplate(ui terminal.UI, realmClient realm.Client, groupID, appID string) (*zip.Reader, []string,  error) {
+	if i.TemplateID == "" {
+		return nil, nil, nil
+	}
+
+	var paths []string
+	if !ui.AutoConfirm() {
+		selectablePaths := []string{
+			frontendPath,
+			backendPath,
+		}
+		if err := ui.AskOne(
+			&paths,
+			&survey.MultiSelect{
+				Message: "Where would you like to export the template?",
+				Options: selectablePaths,
+			}); err != nil {
+			return nil, nil, err
+		}
+	} else {
+		paths = append(paths, frontendPath)
+	}
+
+	compatibleTemplates, err := realmClient.CompatibleTemplates(groupID, appID)
+	if err != nil {
+		return nil, nil, err
+	}
+
+	for _, template := range compatibleTemplates {
+		// This is the ID that they selected
+		if template.ID == i.TemplateID {
+			// Fetch the template
+			templateZip, err := realmClient.ClientTemplate(groupID, appID, template.ID)
+			if err != nil {
+				return nil, nil, err
+			}
+			return templateZip, paths, nil
+		}
+	}
+
+	return nil, nil, fmt.Errorf("templateID %s is not compatible with this app", i.TemplateID)
```

Comment:
try to keep code-like words out of the user-facing messaging
```suggestion
	return nil, nil, fmt.Errorf("template '%s' is not compatible with this app", i.TemplateID)
```